### PR TITLE
[TF] Create common WaitForConfig implementation

### PR DESCRIPTION
**Pull request by nmittler**
Security tests have a custom implementation to deal with issues related to WaitForConfig. Moving to a common implementation.

**Please provide a description of this PR:**

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
LGTM but I think we should make all places that should, in the future, fail if wait fails, use `OrFail` so that we just swap out the code to actually fail if it fails.

As is, and usage of `OrFail` represents a bug/flaky test waiting to happen

**Comment by nmittler** [comment]
@howardjohn PTAL ... `OrFail` now just logs the failure.

**Review by lei-tang — APPROVED** [review]
LGTM.

File: pkg/test/framework/config.go [hunk 1]
```diff
@@ -112,6 +115,40 @@ func (c *configManager) DeleteYAMLOrFail(t test.Failer, ns string, yamlText ...s
...
+	for _, c := range c.ctx.Clusters() {
+		ik, err := istioctl.New(ctx, istioctl.Config{Cluster: c})
+		if err != nil {
+			return err
+		}
+
+		for _, config := range yamlText {
+			config := config
+
+			// TODO(https://github.com/istio/istio/issues/37324): It's currently unsafe
```
Comment (douglas-reid) [inline pkg/test/framework/config.go:129]: this sounds like we could be in for surprises?  How big of a concern is this issue?

**Review by douglas-reid — APPROVED** [review]

File: pkg/test/framework/config.go (hunk 1 above)
Comment (nmittler) [reply pkg/test/framework/config.go:129]: I'm not entirely sure TBH. I just realized it was an issue and raised https://github.com/istio/istio/pull/37325 as a potential mitigation. I suspect fixing istioctl would be a major undertaking and is unlikely to ever happen.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**